internal/service/resources/kesselinventoryservice.go

```diff
@@ -70,24 +70,37 @@ func (c *InventoryService) DeleteResource(ctx context.Context, r *pb.DeleteResou
 		return nil, status.Errorf(codes.Unauthenticated, "failed to get identity: %v", err)
 	}
 
-	reporterResource, err := RequestToDeleteResource(r, identity)
-	if err != nil {
-		log.Error("Failed to build reporter resource ID: ", err)
-		return nil, status.Errorf(codes.InvalidArgument, "failed to build reporter resource ID: %v", err)
-	}
+	if c.useV1beta2Db() {
+		log.Info("Delete Resource using v1beta2db")
+		if reporterResourceKey, err := reporterKeyFromResourceReference(r.GetReference()); err == nil {
+			if err = c.Ctl.Delete(reporterResourceKey); err == nil {
+				return ResponseFromDeleteResource(), nil
+			} else {
+				return nil, err
+			}
+		} else {
+			return nil, err
```

Comment:
We should make sure to return proper status codes from these `return nil, err` branches where it makes sense. By default Kratos will return 500s without any `status.Errorf` specified.